ambry-network/src/main/java/com.github.ambry.network/NetworkClient.java

```diff
@@ -191,17 +215,39 @@ public void close() {
   }
 
   /**
-   * A class that consists of a {@link RequestInfo} with the time at which it is added to the queue.
+   * A class that consists of a {@link RequestInfo} and some metadata related to the request
    */
   private class RequestMetadata {
-    // the time at which this request was queued.
-    long requestQueuedTimeMs;
+    // to track network request related metrics
+    ClientNetworkRequestMetrics clientNetworkRequestMetrics;
     // the RequestInfo associated with the request.
     RequestInfo requestInfo;
+    // the time at which this request was queued.
+    private long requestQueuedTimeMs;
+    // the time at which this request was sent(or moved from queue to in flight state)
+    private long requestSentTimeMs;
 
-    RequestMetadata(long requestQueuedTimeMs, RequestInfo requestInfo) {
+    RequestMetadata(long requestQueuedTimeMs, RequestInfo requestInfo,
+        ClientNetworkRequestMetrics clientNetworkRequestMetrics) {
       this.requestInfo = requestInfo;
       this.requestQueuedTimeMs = requestQueuedTimeMs;
+      this.clientNetworkRequestMetrics = clientNetworkRequestMetrics;
+    }
+
+    /**
+     * Actions to be done on dequeue of this request and ready to be sent
+     */
+    void onRequestDequeue() {
+      requestSentTimeMs = System.currentTimeMillis();
+      clientNetworkRequestMetrics.updateQueueTime(requestSentTimeMs - requestQueuedTimeMs);
+    }
+
+    /**
+     * Actions to be done on receiving response for the request sent
+     */
+    void onResponseReceive() {
+      networkMetrics.requestResponseRoundTripTime.update(System.currentTimeMillis() - requestSentTimeMs);
```

Comment:
Yep. So, we might have to modify ResponseInfo to include these latency values right? do we have a ticket to work on this ?